limacharlie/sync.go

```diff
@@ -0,0 +1,198 @@
+package limacharlie
+
+import (
+	"fmt"
+	"strings"
+)
+
+const (
+	OrgConfigLatestVersion = 3
+)
+
+// Describes which configuration
+// types to Sync.
+type SyncOptions struct {
+	// Force makes the remove Org an exact mirror of the
+	// configuration provided, adding and removing.
+	// Otherwise elements will only be added, not removed.
+	IsForce bool
+
+	// IgnoreInaccessible ignores elements that are
+	// locked and cannot be modified by the credentials
+	// currently in use.
+	IsIgnoreInaccessible bool
+
+	// Only simulate changes to the Org.
+	IsDryRun bool
+
+	SyncDRRules     bool
+	SyncOutputs     bool
+	SyncResources   bool
+	SyncIntegrity   bool
+	SyncFPRules     bool
+	SyncExfil       bool
+	SyncArtifacts   bool
+	SyncNetPolicies bool
+}
+
+type DRRuleName = string
+
+type OrgConfig struct {
+	DRRules map[DRRuleName]CoreDRRule `json:"rules" yaml:"rules"`
+}
+
+type OrgSyncOperation struct {
+	ElementType string `json:"type"`
+	ElementName string `json:"name"`
+	IsAdded     bool   `json:"is_added"`
+	IsRemoved   bool   `json:"is_removed"`
+}
+
+func (org Organization) SyncFetch(options SyncOptions) (OrgConfig, error) {
+	// TODO
```

Comment:
We could return a specific error type for feature not supported yet? Same idea for the TODOs in SyncPush.